### PR TITLE
Browser Cache for images

### PR DESCRIPTION
**Pull request by ChaituVR**

### CODE REVIEW

File: mirror.js [hunk 1]
```diff
@@ -19,16 +19,19 @@ async function handleRequest(event) {
...
     response = await fetch(imageRequest, options);
     response = new Response(response.body, response);
     response.headers.append('Cache-Control', 's-maxage=604800');
     event.waitUntil(cache.put(cacheKey, response.clone()));
+  } else {
+    response = new Response(response.body, response);
   }
-
+  
+  response.headers.set('Cache-Control', 'max-age=31536000');
```
Comment (bonustrack) [inline mirror.js:35]: Why 31536000?

File: mirror.js [hunk 2]
```diff
@@ -19,16 +19,19 @@ async function handleRequest(event) {
...
   if (!response) {
+    const imageRequest = new Request(imageURL, {
+      headers: request.headers
+    });
     response = await fetch(imageRequest, options);
     response = new Response(response.body, response);
     response.headers.append('Cache-Control', 's-maxage=604800');
     event.waitUntil(cache.put(cacheKey, response.clone()));
+  } else {
+    response = new Response(response.body, response);
```
Comment (bonustrack) [inline mirror.js:32]: If you don't add this `response.headers.set` is not working?

File: mirror.js (hunk 1 above)
Comment (ChaituVR) [reply mirror.js:35]: It is just users browser cache.. user can remove cache anytime so added one year.. as suggested in lighthouse docs:
```
The max-age directive tells the browser how long it should cache the resource in seconds. This example sets the duration to 31536000, which corresponds to 1 year: 60 seconds × 60 minutes × 24 hours × 365 days = 31536000 seconds.

When possible, cache immutable static assets for a long time, such as a year or longer.
```
https://web.dev/uses-long-cache-ttl/?utm_source=lighthouse&utm_medium=devtools

File: mirror.js (hunk 2 above)
Comment (ChaituVR) [reply mirror.js:32]: yes correct :P it needs to be in `Response`.. else getting `500 error`..  to debug the error cache doesnt work in local :P so better to have to this like in if condition :)

File: mirror.js (hunk 1 above)
Comment (bonustrack) [reply mirror.js:35]: Isn't response.headers.set going to overwrite the call with response.headers.append before?

**Comment by ChaituVR** [comment]
Can we merge this? Number of image calls reduced?

**Pull request closed without merging**